src/globals.c

```diff
@@ -412,6 +414,36 @@ bool set_default_urls()
 	return true;
 }
 
+bool set_non_interactive_option(char *option)
+{
+	bool ret;
+	char *option_lower = NULL;
+
+	if (!option) {
+		error("Option shouldn't be NULL, please choose either 'yes' or 'no' as your option\n");
+		return false;
+	}
+
+	option_lower = str_tolower(option);
+
+	if (strcmp(option_lower, "y") == 0 || strcmp(option_lower, "yes") == 0) {
+		globals.interactive = false;
+		globals.non_interactive_yes = true;
+		ret = true;
+	} else if (strcmp(option_lower, "n") == 0 || strcmp(option_lower, "no") == 0) {
+		globals.interactive = false;
```

Comment:
So if you pass y or n interactive is set to false. Seems like a typo? Also having the option being a negative seems a little awkward. Maybe have the option be interactive instead?

Reply:
interactive = true is the default value, so if the user doesn't use `--non-interactive` swupd will interactively ask for either 'n' or 'y'. But when the user chooses non interactive mode by using the flag they can choose `--non-interactive=yes` or `--non-interactive=no`, in either case it is non interactive, the first case is just saying "respond Yes to all prompts", the second case is saying "respond No to all prompts".

Reply:
Ah that's interesting, I didn't understand the man page change very well clearly X(. Seems fine to me.

Reply:
do you have suggestions on how to make the wording for the --help menu and the man pages more clear? I have the concern of people getting confused with this flag now 🤔 

This is what I'm showing now in the help menu:
`--non-interactive=[yes/no] Sets an automatic response to prompts to run non-interactively`

And the same in the Man page:
```
 • --non-interactive={yes|no}

         Sets an automatic response to prompts to run non-interactively.
```